File: 0006-zigzag-conversion/0006-zigzag-conversion.java
class Solution {
    public String convert(String s, int n) {
       
        int si = s.length();

         if (n == 1 || n >= si) {
            return s;
        }

        int j = 0;
        int dir = -1;

        String[] v = new String[n];
        for (int i = 0; i < n; i++) {
            v[i] = "";
        }

        for (int i = 0; i < si; i++) {
            if (j == 0 || j == n - 1) {
                dir *= -1;
            }
            v[j] += s.charAt(i);
            if (dir == 1) {
                j++;
            } else {
                j--;
            }
        }

        StringBuilder ans = new StringBuilder();
        for (String e : v) {
            ans.append(e);
        }
        return ans.toString();
    }
}

File: 0006-zigzag-conversion/0006-zigzag-conversion.cpp
class Solution {
public:
    string convert(string s, int n) {

        if (n <= 1) {
            return s;
        }
        vector<string> v(n, "");

        int j = 0, dir = -1;

        for (int i = 0; i < s.length(); i++) {
            if (j == n - 1  || j == 0) {
                dir *= (-1);
            }
            v[j] += s[i];
            if (dir == 1) {
                j++;
            } else {
                j--;
            }
        }
        string res;
        for (auto it : v) {
            res += it;
        }
        return res;
    }
};
